ear_appraisal_test.go

```diff
@@ -19,7 +20,7 @@ func TestAppraisalExtensions_SetGetKeyAttestation_ok(t *testing.T) {
 		},
 	}
 
-	kp, err := ecdsa.GenerateKey(elliptic.P256(), new(zeroSource))
+	kp, err := keygen.ECDSALegacy(elliptic.P256(), new(zeroSource))
```

Comment:
I am not against the fix as such:

However, is it really needed ? Should we not seek feedback on the issue with ECDSA Library maintainers! Then we should not rely on third party library as a workaround?

Reply:
If you want to make the case for reverting a change in the Go standard library I am certainly not stopping you :-D

From a personal/pragmatic point of view, I am content with this minimal-impact fix.

Reply:
Yes, raised an issue:
https://github.com/golang/go/issues/69248

Please review, if you want to add anything on it!

Reply:
Also if the new change is by `design` should we not use a Non-Zero Reader in our code to avoid crash, instead of fundamentally by-pass the core golang module ??

We will soon become stale using the library, which we are un-aware of?

Reply:
The test _needs_ to construct a key deterministically.

Reply:
> The test _needs_ to construct a key deterministically.

yes, what I meant was we can supply a constant deterministic byte array to the input rather than always `zero`

Reply:
Ah, sure.

Reply:
> We will soon become stale using the library, which we are un-aware of?
> 
> I mean `filippo.io/keygen` here:

No, `filippo.io/keygen` is forever stable - that's the promise.

Instead, if we use `GenerateKey()`, golang makes no promises about the stability of the output.  From the [`ecdsa` package documentation](https://pkg.go.dev/crypto/ecdsa#GenerateKey):

> _"Note that the returned key does not depend deterministically on the bytes read from rand, and **may change between calls and/or between versions.**"_